api/v1alpha1/spire_server_config_types.go

```diff
@@ -165,10 +171,182 @@ type CASubject struct {
 	CommonName string `json:"commonName,omitempty"`
 }
 
+// UpstreamAuthority defines the configuration for the upstream certificate authority
+// that SPIRE Server will use to obtain its signing certificate.
+// It supports different plugins such as SPIRE, Vault, and cert-manager for upstream CA integration.
+type UpstreamAuthority struct {
+	// Type specifies the type of upstream authority plugin to use.
+	// Allowed values: "spire", "vault", "cert-manager".
+	// It determines which one of the optional configurations below should be populated.
+	Type string `json:"type,omitempty"`
```

Comment:
Would it be feasible to not rely on an explicit `spec.upstreamAuthority.type` to indicate which type users are using? If the different types are mutually exclusive, this field seems like a redundant input we're asking from users.

Reply:
> the user gives multiple fields as input, which should take precedence?

Would it be a valid usecase? In what situations would a user configure multiple types, given that we can only enable one at a time?

If I try to set multiple types in upstream Helm charts, it will throw an error:

```
Error: execution error at (spire/charts/spire-server/templates/server-resource.yaml:4:19): You can only enable a single Upstream Authority.
```

Maybe we should expect and restrict user to set only one at a time, and deactivate other types.